From 09dad74181806895a40e749464f84ccc027b4fe6 Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@gnome.org>
Date: Fri, 16 May 2014 14:16:43 +0200
Subject: [PATCH 2/2] 730243 Port to vte-2.91 API

       * configure.ac: Build against vte-2.91.
       * src/uicommon/nmv-terminal (init_body): Use
       vte_terminal_set_font instead of
       vte_terminal_set_font_from_string.
       (init_pty): vte_terminal_set_pty expects a VtePty instead of a
       fd.  Adjust according to that.
---
 configure.ac                 |  4 ++--
 src/uicommon/nmv-terminal.cc | 14 ++++++++++++--
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 04f5a26..272acd9 100644
--- configure.ac
+++ configure.ac
@@ -361,7 +361,7 @@ dnl library dependencies for the nemiver common UI module
 DEP_UICOMMON=" gtkmm-3.0 >= $LIBGTKMM_VERSION \
                gtk+-3.0 >= $LIBGTK_VERSION \
                gtksourceviewmm-3.0 >= $LIBGTKSOURCEVIEWMM_VERSION \
-               vte-2.90 >= $LIBVTE_VERSION"
+               vte-2.91 >= $LIBVTE_VERSION"
 
 PKG_CHECK_MODULES(NEMIVERUICOMMON, $DEP_UICOMMON $DEP_COMMON $DEP_MEMORYVIEW)
 NEMIVERUICOMMON_CFLAGS="$NEMIVERUICOMMON_CFLAGS $CPPUNIT_CFLAGS"
@@ -384,7 +384,7 @@ AC_SUBST(NEMIVERWORKBENCH_CFLAGS)
 
 dnl library dependencies for the nemiver debug perspective plugin
 DEP_PERSP="gtksourceviewmm-3.0 >= $LIBGTKSOURCEVIEWMM_VERSION \
-           vte-2.90 >= $LIBVTE_VERSION"
+           vte-2.91 >= $LIBVTE_VERSION"
 
 PKG_CHECK_MODULES(NEMIVERDBGPERSP, $DEP_UICOMMON $DEP_VFS $DEP_PERSP $DEP_MEMORYVIEW $DEP_DYNAMICLAYOUT)
 NEMIVERDBGPERSP_LIBS="$NEMIVERDBGPERSP_LIBS $CPPUNIT_LIBS"
diff --git a/src/uicommon/nmv-terminal.cc b/src/uicommon/nmv-terminal.cc
index 717c094..e7432c1 100644
--- src/uicommon/nmv-terminal.cc
+++ src/uicommon/nmv-terminal.cc
@@ -48,11 +48,13 @@
 #include <gtkmm/menu.h>
 #include <gtkmm/builder.h>
 #include <gtkmm/uimanager.h>
+#include <pangomm/fontdescription.h>
 #include <vte/vte.h>
 #include <glib/gi18n.h>
 #include "common/nmv-exception.h"
 #include "common/nmv-log-stream-utils.h"
 #include "common/nmv-env.h"
+#include "common/nmv-error.h"
 #include "nmv-ui-utils.h"
 
 NEMIVER_BEGIN_NAMESPACE(nemiver)
@@ -136,7 +138,8 @@ struct Terminal::Priv {
         THROW_IF_FAIL (vte);
 
         // Mandatory for vte 0.14	
-        vte_terminal_set_font_from_string (vte, "monospace");
+        Pango::FontDescription font_desc ("monospace");
+        vte_terminal_set_font (vte, font_desc.gobj());
 
         vte_terminal_set_scroll_on_output (vte, TRUE);
         vte_terminal_set_scrollback_lines (vte, 1000);
@@ -266,7 +269,14 @@ struct Terminal::Priv {
         THROW_IF_FAIL (slave_pty);
         THROW_IF_FAIL (master_pty);
 
-        vte_terminal_set_pty (vte, master_pty);
+        GError *err = 0;
+        VtePty *pty = vte_pty_new_foreign_sync (master_pty, 0, &err);
+        GErrorSafePtr error (error);
+        THROW_IF_FAIL2 (!error, error->message);
+
+        vte_terminal_set_pty (vte, pty);
+        g_object_unref (pty);
+
         return true;
     }
 };//end Terminal::Priv
-- 
1.9.0

